Homework 7/ParseTree/ParseTree.c

```diff
@@ -0,0 +1,178 @@
+#pragma warning(disable: 4996 6031 6066)
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../ParseTree/ParseTree.h"
+
+Node* createNode()
+{
+    return calloc(1, sizeof(Node));
+}
+
+void deleteNode(Node* node)
+{
+    free(node);
+}
+
+void deleteParseTree(Node* root)
+{
+    if (root == NULL)
+    {
+        return;
+    }
+    if (root->leftChild != NULL)
+    {
+        deleteParseTree(root->leftChild);
+    }
+    if (root->rightChild != NULL)
+    {
+        deleteParseTree(root->rightChild);
+    }
+    deleteNode(root);
+}
+
+bool isOperatorInString(char symbol)
+{
+    return symbol == '+' || symbol == '*' || symbol == '/';
+}
+
+bool isNumberInString(char symbol)
+{
+    return symbol != '(' && symbol != ')' && symbol != ' ';
+}
+
+Node* createNewNodeForParseTree(char* stringOfNumbersAndOperators, int* index)
+{
+    ++(*index);
+    const int lengthOfString = (int)strlen(stringOfNumbersAndOperators);
+    Node* newNode = createNode();
+    while (*index < lengthOfString
+        && !isNumberInString(stringOfNumbersAndOperators[*index]))
+    {
+        ++(*index);
+    }
+    if (isOperatorInString(stringOfNumbersAndOperators[*index])
+        || (stringOfNumbersAndOperators[*index] == '-'
+            && stringOfNumbersAndOperators[*index + 1] == ' '))
+    {
+        newNode->symbol = stringOfNumbersAndOperators[*index];
+        newNode->leftChild = createNewNodeForParseTree(stringOfNumbersAndOperators,
+            index);
+        newNode->rightChild = createNewNodeForParseTree(stringOfNumbersAndOperators,
+            index);
+    }
+    else if (isNumberInString(stringOfNumbersAndOperators[*index]))
+    {
+        char stringForNumber[30] = { '\0' };
+        int indexOfStringForNumber = 0;
+        while (isNumberInString(stringOfNumbersAndOperators[*index])
+            && !isOperatorInString(stringOfNumbersAndOperators[*index]))
+        {
+            stringForNumber[indexOfStringForNumber] = stringOfNumbersAndOperators[*index];
+            ++indexOfStringForNumber;
+            ++(*index);
+        }
+        --(*index);
+        int number = atoi(stringForNumber);
+        newNode->number = number;
+    }
+    return newNode;
+}
+
+void addToString(char* string, int* indexForString, char symbolToAdding, bool spaceOption)
+{
+    string[*indexForString] = symbolToAdding;
+    ++(* indexForString);
```

Comment:
```suggestion
    ++(*indexForString);
```
Унарные операторы пробелом не выделяются